Sources/Kitura/RouterRequest.swift

```diff
@@ -234,42 +225,46 @@ public class RouterRequest {
 }
 
 private class Cookies {
-
-    /// Storage of parsed Cookie headers
-    fileprivate var cookies = [String: HTTPCookie]()
-
-    /// Static for Cookie header key value
-    private let cookieHeader = "cookie"
-
-    fileprivate init(headers: HeadersContainer) {
-        guard let rawCookies = headers[cookieHeader] else {
-            return
+    private static var separator: RegularExpressionType = {
+        do {
+            // matches that do not contain semicolons and do not start with whitespaces
+            // effectively splits string by ";\\s*"
+            return try RegularExpressionType(pattern: "[^;\\s][^;]*", options: [])
+        } catch { // should never throw here, famous last words
+            Log.error("Error creating cookie separator regex: \(error)")
+            exit(1)
```

Comment:
OK, maybe I'm going mad here, but just because it's `static` doesn't mean it's only computed once. Every time the property is accessed it is recomputed.

Try running this:

```swift
class Test {
    static var aString: String {
        print("test")
        return "string"
    }
}

print(Test.aString)
print(Test.aString)
```

It prints:

```
test
string
test
string
```